include/sentry.h

```diff
@@ -1295,17 +1295,64 @@ SENTRY_EXPERIMENTAL_API void sentry_transaction_context_remove_sampled(
  * Starts a new Transaction based on the provided context, restored from an
  * external integration (i.e. a span from a different SDK) or manually
  * constructed by a user.
+ *
+ * `sentry_transaction_finish` should be called after this is invoked, otherwise
+ * the Transaction will not be sent to sentry. New spans cannot be created
+ * unless there exists an active Transaction.
  */
 SENTRY_EXPERIMENTAL_API void sentry_transaction_start(
     sentry_value_t transaction_context);
 
 /**
- * Finishes and sends the current transaction to sentry. The event ID of the
- * transaction will be returned if this was successful; A nil UUID will be
- * returned otherwise.
+ * Finishes and sends the current active Transaction to sentry. Any unfinished
+ * spans are removed from the Transaction before it is sent over.
+ *
+ * No new spans can be created after this is invoked unless a new Transaction is
+ * started via `sentry_transaction_start`.
  */
 SENTRY_EXPERIMENTAL_API sentry_uuid_t sentry_transaction_finish();
 
+/**
+ * Starts a new Span.
+ *
+ * If `parent_span` is `sentry_value_null`, then the current active Transaction
+ * is used as the parent for the new Span. An active Transaction must be created
+ * via `sentry_transaction_start` in order for the Span to be successfully
+ * created.
+ *
+ * If `parent_span` is another Span, it must belong to the current active
+ * Transaction in order for Span creation to succeed. This will take ownership
+ * of any `parent_span`s that do reference non-existent Spans in the current
+ * active Transaction.
```

Comment:
i can't really parse this paragraph...  I'd expect that the parent span is only borrowed for this function call because the caller will still have to pass the parent to it's own finish call.

Reply:
ownership of `parent_span` is stolen if it logically should not exist anymore: for example, if for some reason the `parent_span` was not finished before its parent Transaction was finished, there is no point letting the user finish the `parent_span` nor is there any point in letting more children be created from it, as all of its descendants will never be reported to sentry. 

the idea was to steal ownership and to discard the `parent_span` to prevent the user from performing more no-ops with an invalid span.